Add target for CY8CPROTO-062S3-4343W

kyle-cypress opened the pull request:
<!--
For more information on the requirements for pull requests, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->
### Description (*required*)
Add target for Cypress board CY8CPROTO-062S3-4343W
This depends on PR #11884 
<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->
##### Summary of change (*What the change is for and why*)


##### Documentation (*Details of any document updates required*)

----------------------------------------------------------------------------------------------------------------
### Pull request type (*required*)

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [x] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results (*required*)
[GT_FT_PROTO_062S3_4343W_GCC_ARM.txt](https://github.com/ARMmbed/mbed-os/files/3862362/GT_FT_PROTO_062S3_4343W_GCC_ARM.txt)
NOTE: The kvstore failures are due to #11879. The sleep failures are due to a known incompatibility between the sleep tests and our UART driver.

[GT_FT_PROTO_062S3_4343W-NET.txt](https://github.com/ARMmbed/mbed-os/files/3865525/GT_FT_PROTO_062S3_4343W-NET.txt)
NOTE: The failures are known issues with the integration test suite which we recently enabled. They apply across Cypress targets and will be fixed in a future PR.
<!--
    Required
    For example, add test results for new target
-->
    [] No Tests required for this change (E.g docs only update)
    [x] Covered by existing mbed-os tests (Greentea or Unittest)
    [x] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers (*optional*)
@ARMmbed/team-cypress
<!--
    Optional
    Request additional reviewers with @username
-->

----------------------------------------------------------------------------------------------------------------
### Release Notes (*required for feature/major PRs*)

<!--
    All 3 sections are compulsory for Major PR types. For Feature PRs only the summary section is required.
    This section is automatically added to release notes. Please fill in each sub-section with sufficient detail for a user.
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types). 
-->

##### Summary of changes

##### Impact of changes

##### Migration actions required

Commit:
Add target for CY8CPROTO-062S3-4343W

bulislaw commented on targets/targets.json:
```diff
@@ -9123,6 +9123,28 @@
+    "CY8CMOD_062S3_4343W": {
+        "inherits": ["MCU_PSOC6_M4"],
+        "features": ["BLE"],
+        "components_add": ["WHD", "4343W", "CYW43XXX"],
```
What are these components? Are those drivers for external HW? If not they shouldn't be in components.

kyle-cypress replied:
Yes, these are drivers for the external Wifi and BT radio, which were changed to a component in #11840 .

0xc0170 replied:
Also `CYW43XXX` ?

kyle-cypress replied:
Yes, all three are related to external radio support.

Commit:
Fix missing #include in wiced_resource.h

The `CY_PRAGMA` macro is used in certain build scenarios with IAR,
but it is not defined without including "cy_syslib.h"
